framework/src/ics/BoundingBoxIC.C

```diff
@@ -46,16 +50,30 @@ BoundingBoxIC::BoundingBoxIC(const InputParameters & parameters)
     _inside(getParam<Real>("inside")),
     _outside(getParam<Real>("outside")),
     _bottom_left(_x1, _y1, _z1),
-    _top_right(_x2, _y2, _z2)
+    _top_right(_x2, _y2, _z2),
+    _int_width(getParam<Real>("int_width"))
 {
 }
 
 Real
 BoundingBoxIC::value(const Point & p)
 {
-  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
-    if (p(i) < _bottom_left(i) || p(i) > _top_right(i))
-      return _outside;
+  if (_int_width <= 0.0)
+  {
+    for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+      if (p(i) < _bottom_left(i) || p(i) > _top_right(i))
+        return _outside;
+
+    return _inside;
+  }
+  else
+  {
+    Real f_in = 1.0;
+    for (unsigned int i = 0; i < LIBMESH_DIM; ++i)
+      if (_bottom_left(i) != _top_right(i))
+        f_in *= 0.5 * (std::tanh(2.0 * (p(i) - _bottom_left(i)) / _int_width) -
+                       std::tanh(libMesh::pi * (p(i) - _top_right(i)) / _int_width));
```

Comment:
I think this expression needs further explanation. Why the asymmetry between the two sub terms (one has a 2.0 the other one a pi).

Reply:
Oops. Thanks for catching that.